thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/api/MetricTimeSeries.java

```diff
@@ -41,110 +45,108 @@ public MetricSchema getSchema() {
-        break;
-      case LONG:
-        ret = buffer.getLong();
-        break;
-      case FLOAT:
-        ret = buffer.getFloat();
-        break;
-      case DOUBLE:
-        ret = buffer.getDouble();
-        break;
+  /**
+   * Gets the metric value with the timestamp if the value exists; otherwise, the default number is returned.
+   *
+   * @param timeWindow the timestamp.
+   * @param name the metric name.
+   * @param defaultNumber the default number for the returned value if the target value does not exist.
+   *
+   * @return the metric value with the timestamp if the value exists; otherwise, the default number is returned.
+   */
+  public Number getOrDefault(long timeWindow, String name, Number defaultNumber) {
+    Number ret = defaultNumber;
+
+    boolean[] hasValueBuffer = hasValue.get(timeWindow);
+    if (hasValueBuffer != null && hasValueBuffer[schema.getMetricIndex(name)]) {
+      ByteBuffer buffer = metricsValue.get(timeWindow);
+      if (buffer != null) {
+        buffer = buffer.duplicate();
+        buffer.position(schema.getOffset(name));
+        MetricType metricType = schema.getMetricType(name);
+        ret = NumberUtils.readFromBuffer(buffer, metricType);
       }
     }
+
     return ret;
   }
 
+  /**
+   * Increments the metric value of the timestamp with delta.
+   *
+   * @param timeWindow the timestamp.
+   * @param name the metric name.
+   * @param delta the non-null value to be added to the metric value.
+   */
   public void increment(long timeWindow, String name, Number delta) {
-    initBufferForTimeWindow(timeWindow);
-    ByteBuffer buffer = timeseries.get(timeWindow);
-    Number newValue;
+    Number newValue = delta;
+    Number oldValue = getOrDefault(timeWindow, name, null);
 
-    if (buffer != null) {
-      // get Old Value
-      Number oldValue = get(timeWindow, name);
+    if (oldValue != null) {
       MetricType metricType = schema.getMetricType(name);
-      switch (metricType) {
-      case SHORT:
-        newValue = oldValue.shortValue() + delta.shortValue();
-        break;
-      case INT:
-        newValue = oldValue.intValue() + delta.intValue();
-        break;
-      case LONG:
-        newValue = oldValue.longValue() + delta.longValue();
-        break;
-      case FLOAT:
-        newValue = oldValue.floatValue() + delta.floatValue();
-        break;
-      case DOUBLE:
-        newValue = oldValue.doubleValue() + delta.doubleValue();
-        break;
-      default:
+      newValue = NumberUtils.sum(newValue, oldValue, metricType);
+      if (newValue == null) {
         throw new UnsupportedOperationException(
             "unknown metricType:" + metricType + " for column:" + name);
       }
-    } else {
-      newValue = delta;
     }
+
     set(timeWindow, name, newValue);
   }
 
   public void aggregate(MetricTimeSeries series) {
-    for (long timeWindow : series.timeseries.keySet()) {
+    for (long timeWindow : series.metricsValue.keySet()) {
```

Comment:
Is these aggregate metric actually used? It is a cross metric aggregation. Wondering where it could be used.